boto3/s3/transfer.py

```diff
@@ -122,499 +122,71 @@ def __call__(self, bytes_amount):
-
-class TransferConfig(object):
     def __init__(self,
                  multipart_threshold=8 * MB,
                  max_concurrency=10,
                  multipart_chunksize=8 * MB,
                  num_download_attempts=5,
                  max_io_queue=100):
-        self.multipart_threshold = multipart_threshold
-        self.max_concurrency = max_concurrency
-        self.multipart_chunksize = multipart_chunksize
-        self.num_download_attempts = num_download_attempts
-        self.max_io_queue = max_io_queue
+        super(TransferConfig, self).__init__(
+            multipart_threshold=multipart_threshold,
+            max_request_concurrency=max_concurrency,
+            multipart_chunksize=multipart_chunksize,
+            num_download_attempts=num_download_attempts,
+            max_io_queue_size=max_io_queue
+        )
+        # Some of the argument names are not the same as the inherited
+        # S3TransferConfig so we add aliases so you can still access the
+        # old version of the names.
+        for alias in self.ALIAS:
+            setattr(self, alias, getattr(self, self.ALIAS[alias]))
+
+    def __setattr__(self, name, value):
+        # If the alias name is used, make sure we set the name that it points
+        # to as that is what actually is used in governing the TransferManager.
+        if name in self.ALIAS:
+            super(TransferConfig, self).__setattr__(self.ALIAS[name], value)
+        # Always set the value of the actual name provided.
+        super(TransferConfig, self).__setattr__(name, value)
 
 
 class S3Transfer(object):
-
-    ALLOWED_DOWNLOAD_ARGS = [
-        'VersionId',
-        'SSECustomerAlgorithm',
-        'SSECustomerKey',
-        'SSECustomerKeyMD5',
-        'RequestPayer',
-    ]
-
-    ALLOWED_UPLOAD_ARGS = [
-        'ACL',
-        'CacheControl',
-        'ContentDisposition',
-        'ContentEncoding',
-        'ContentLanguage',
-        'ContentType',
-        'Expires',
-        'GrantFullControl',
-        'GrantRead',
-        'GrantReadACP',
-        'GrantWriteACL',
-        'Metadata',
-        'RequestPayer',
-        'ServerSideEncryption',
-        'StorageClass',
-        'SSECustomerAlgorithm',
-        'SSECustomerKey',
-        'SSECustomerKeyMD5',
-        'SSEKMSKeyId',
-    ]
+    ALLOWED_DOWNLOAD_ARGS = TransferManager.ALLOWED_DOWNLOAD_ARGS
+    ALLOWED_UPLOAD_ARGS = TransferManager.ALLOWED_UPLOAD_ARGS
 
     def __init__(self, client, config=None, osutil=None):
-        self._client = client
         if config is None:
             config = TransferConfig()
-        self._config = config
         if osutil is None:
             osutil = OSUtils()
-        self._osutil = osutil
+        self._manager = TransferManager(client, config, osutil)
+
+    @classmethod
+    def from_transfer_manager(cls, transfer_manager):
+        """Instantiate S3Transfer from s3transfer.TransferManager instance"""
+        cls_instance = super(S3Transfer, cls).__new__(cls)
```

Comment:
I can do that. It is basically this:
```py
import boto3
from boto3.s3.transfer import S3Transfer
from s3transfer.manager import TransferManager

c = boto3.client('s3')
manager = TransferManager(c)
transfer = S3Transfer.from_transfer_manager(manager)
```

I actually wanted opinions on this. I though it was weird that I shortcutted the initializer to instantiate an internal property. But that was the cleanest way to inject a TransferManger other than making a public ``self.manager`` on ``S3Transfer`` class which I would be fine with as well. If I allowed them to pass a transfer manager to the initalizer of ``S3Transfer`` I would have to make client marked as optional (since the client is required already by the ``TransferManager``) which was strange as well.

What do you think?